### ISSUE TITLE
Decongesting Pull Requests

### ISSUE THREAD

**Issue by masvelio** [hicommonwealth/commonwealth#7376]
## Description

We, as a team, are struggling with doing code review on regular, daily basis. We need 2 approvals on each PR to allow author to merge the pull request. We use `reviewers` field in PR,  so other engineers are notified about review request. However, it does not seem to work. We have PRs open for weeks which leads to many problems like:
- resolving conflicts
- stale PRs
- polluted board with `In progress` tasks
- delays in delivery
- and more

## Possible solutions
- change number of approvals from 2 to 1 (disclaimer: we should still have 2 approvals for features that have changes in more than one "package" eg front and back or front and protocol)
- start monitoring "time to merge" and other metrics that would help us merging quicker
  -  [github solution to explore](https://github.blog/2023-07-19-metrics-for-issues-pull-requests-and-discussions/)
  - [third party solution to explore](https://gitdailies.com/)
  - others?
- hook up some slack bot that will keep reminding about open PRs
   - [github solution to explore](https://github.blog/2020-04-21-stay-on-top-of-your-code-reviews-with-scheduled-reminders/) 
   - [third party solution to explore](https://codekickbot.com/#product)
   - others?

**Comment by burtonator** [comment]
I definitely agree here.  I think what we're seeing is that PRs are snowballing. 

Do others on the team feel the same way?

If the PR acquisition rate is slow then it means users are encouraged to create larger PRs since they can't get their code merged thereby creating even larger PRs. That in turn makes it difficult for users to review them (and causes the potential for poor reviews). This further discourages users to review since PRs are now huge causing it to snowball more.

I think we could fix this by going down to 1 reviewer and implementing "code owners" 

https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners

Basically it's a file in the root that takes the paths of code and contains a list of owners of that code that can review.

For example, backend code could have code owners for backend devs and the same for frontend.  We'd also put blockchain people on blockchain code.  

This way all PRs would require one *quality* reviewer and then we could merge.

**Comment by burtonator** [comment]
I also agree regarding KPIs. I was looking into that too.

I think we should try to close some of the old draft PRs so that they're not stuck around forever too.

**Comment by burtonator** [comment]
Also, There are a lot of tools that have dashboards as @masvelio linked above.

They all seem to cost some amount of money like $50-100 per month but they seem like they have a ton of features.  We'd have to spend some time to review.

**Comment by Rotorsoft** [comment]
@burtonator are we still pursuing this?

**Comment by Rotorsoft** [comment]
Won't do
